src/algorithm/binarySearchTree.ts

```diff
@@ -182,13 +170,53 @@ export class BinarySearchTree<Key extends number | string, Val> {
 
 	/**
 	 * Prints binary search tree
+	 * Inspired by https://flaviocopes.com/golang-data-structure-binary-search-tree/
 	 * @returns
 	 */
 	public print() {
-		if (this.root != null) {
-			return this.root.toString()
+		// Draw lines for debugging
+		var maxHeight = this.getMaxHeight()
+		if (maxHeight == 0) return
+		var line = "--------"
+		var current = ""
+		for (var i = 0; i < maxHeight; i++) {
+			current += line
+		}
+
+		console.log(current)
+		this.stringify(this.root, 0)
+		console.log(current)
+	}
+
+	private stringify(n: TreeNode<Key, Val> | null, level: number) {
+		if (!n) return
+
+		var format = ""
+		for (var i = 0; i < level; i++) {
+			format += "       "
 		}
 
-		return ""
+		format += "----[ "
+		level++
+
+		// change the order for logs to display smaller value to right
+		this.stringify(n.right, level)
+		console.log(format + n.key)
+		this.stringify(n.left, level)
+	}
+
+	/**
+	 * Return max height of Binary Search Tree
+	 * @returns max height
+	 */
+	public getMaxHeight(): number {
+		return this.getMaxHeightRecursively(this.root)
```

Comment:
not sure I like the signature 🤔 maybe a bit too verbose

what do you think of just using an accessor ?
```ts
public get height(): number {
```
so we could use is like `arr.length` and have `bst.height`

the user doesnt really need to know it's calling a function ^^